File: Cap 04/Topico 4.5/Topico 4.5.2/Decisao Encadeada/decencad.pas
{*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 * 
 * Capitulo ..: 4 - Topico 4.5.2
 * 
 * Programa ..: OPERACAO COM DECISAO ENCADEADA
 *}

program REAJUSTA_SALARIO;
var
  SA, NS: real;
begin
  writeln('DECISAO ENCADEADA - SALARIO');
  writeln;
  write('Entre salario atual: ');
  readln(SA);
  if (SA < 500) then
  begin
    NS := SA * 1.15;
  end
  else
  begin
    if (SA <= 1000) then
    begin
      NS := SA * 1.10;
    end
    else
    begin
      NS := SA * 1.05;
    end;
  end;
  writeln('Novo salario: ', NS:10:2);
end.

File: Cap 04/Topico 4.5/Topico 4.5.2/Decisao Encadeada/decencad.cs
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 4 - Topico 4.5.2
 * 
 * Programa ..: OPERACAO COM DECISAO ENCADEADA
*/

using System;

class Program
{
	
  static float SA, NS;
  static void Main()
  {
	  
    Console.WriteLine("DECISAO ENCADEADA - SALARIO\n");
    
    Console.Write("Entre salario atual: ");
    SA = float.Parse(Console.ReadLine());
    
    if (SA < 500)
    {
      NS = SA * 1.15f;
    }
    else
    {
      if (SA <= 1000)
      {
        NS = SA * 1.10f;
      }
      else
      {
        NS = SA * 1.05f;
      }
    }
    
    Console.WriteLine("Novo salario: {0,10:F2}", NS);
    
  }
}
